Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php

```diff
@@ -946,20 +939,54 @@ private function whenDimensionShineThroughWasAdded(DimensionShineThroughWasAdded
              :newDimensionSpacePointHash AS dimensionspacepointhash,
               h.contentstreamid
             FROM
-                ' . $this->tableNames->hierarchyRelation() . ' h
+                {$this->tableNames->hierarchyRelation()} h
                 WHERE h.contentstreamid = :contentStreamId
-                AND h.dimensionspacepointhash = :sourceDimensionSpacePointHash',
-            [
+                AND h.dimensionspacepointhash = :sourceDimensionSpacePointHash
+        SQL;
+        try {
+            $this->dbal->executeStatement($insertHierarchyRelationsStatement, [
                 'contentStreamId' => $event->contentStreamId->value,
                 'sourceDimensionSpacePointHash' => $event->source->hash,
                 'newDimensionSpacePointHash' => $event->target->hash,
-            ]
-        );
+            ]);
+        } catch (DbalException $e) {
+            throw new \RuntimeException(sprintf('Failed to insert hierarchy relations: %s', $e->getMessage()), 1716490758, $e);
+        }
+    }
+
+    private function whenNodeAggregateWasMoved(NodeAggregateWasMoved $event): void
```

Comment:
btw: I placed the new `when*` methods randomly after the existing ones and there are unrelated methods in between.
I'll sort them alphabetically in a follow-up